### PR TITLE
[fields] Remove `enableAccessibleFieldDOMStructure` behavior

### PR DESCRIPTION
**Pull request by LukasTy**
## Summary

Removes the `enableAccessibleFieldDOMStructure` prop from all pickers and field components. The accessible DOM structure (section-based `PickersTextField`) introduced in v7 is now the only supported mode — the legacy v6 plain `<input>` fallback is no longer available.

### What changed

- Deleted `useFieldV6TextField` hook and all v6 legacy field rendering code
- Removed `TEnableAccessibleFieldDOMStructure` generic parameter from `PickerManager`, all field/picker types, hooks, and components
- Simplified core field hooks (`useField`, `useFieldState`, `buildSectionsFromFormat`, `getSectionOrder`) by removing all conditional v6/v7 branching
- Inlined `useFieldV7TextField` into `useField` and `useFieldRootHandleKeyDown` into `useFieldRootProps` (no longer need separate files)
- Removed `createDateStrForV6InputFromSections` / `getV6InputValueFromSections` from value managers
- Renamed `getV7HiddenInputValueFromSections` → `getHiddenInputValueFromSections`
- Removed `UseDateManagerParameters` and `UseDateTimeManagerParameters` interfaces (were empty after prop removal)
- Deleted e2e fixtures and docs for `NonAccessibleDOMStructure` variants
- Renamed `BrowserV7*` and `MaterialV7*` demo files to drop "V7" suffix
- Renamed `expectFieldValueV7` → `expectFieldValue` and `v7Response` → `response` in test utilities
- Added deprecation notices in v6 and v7 migration guides
- Added `remove-enable-accessible-field-dom-structure` codemod (v9.0.0) that strips the prop from all JSX usages; included in `preset-safe`
- Updated v8 migration guide with codemods section and removed types documentation

### CODE REVIEW

**Comment by LukasTy** [comment]
# Claude Opus 4.6 review

**Overall assessment:** The core refactor is solid. The v6 code path, type parameters, and conditional logic have been properly removed. No functional bugs found. The issues below are cleanup items that should be addressed to complete the simplification.

---

### Issues to fix

#### 1. Stale TODO comment — now actionable

useFieldRootProps.ts:
```ts
// TODO: Inline onContainerKeyDown once the old DOM structure is removed
```
The old DOM structure has been removed. This TODO should either be acted upon (inline the handler) or removed.

#### 2. Outdated JSDoc comments referencing "non-accessible DOM structure"

Four files still contain `It is not used by the non-accessible DOM structure (with an <input /> element for editing).` — this DOM structure no longer exists. These comments should be updated to just describe what the hook does:

- useFieldRootProps.ts
- useFieldHiddenInputProps.ts
- useFieldSectionContainerProps.ts
- useFieldSectionContentProps.ts

#### 3. Outdated comment in PickerFieldUI — the "future work" is now possible

PickerFieldUI.tsx:
```ts
/**
 * ...Once the non-accessible DOM structure will be removed, we will be able to
 * remove the `textField` slot and clean this logic.
 */
```
The non-accessible DOM structure **was** removed by this PR. Either follow through on the cleanup or update the comment to explain why the `textField` slot is still needed.

#### 4. `useFieldV7TextField` naming — no longer version-specific

useFieldV7TextField.ts is the only implementation now, and useField.ts is a trivial pass-through:
```ts
export const useField = (...) => {
  return useFieldV7TextField(parameters);
};
```
Consider either inlining `useFieldV7TextField` into `useField` or renaming the file/function to drop "V7". The indirection is unnecessary now.

#### 5. Duplicate error code in error-codes.json

The error about `sectionListRef` not being initialized now exists in two codes:
- error-codes.json — old message, still suggests `enableAccessibleFieldDOMStructure={false}` (a prop that no longer exists)
- error-codes.json — new message, correctly updated

Per the project guidelines: *"If the update created a new error code, but the new and original message have the same number of arguments and semantics haven't changed, update the original error in error-codes.json instead of creating a new code."* The semantics did change (removed one paragraph), so having a new code (205) is arguably correct for backward compatibility. But it's worth verifying this is intentional. The old code 159 will still be served to users hitting the error on older versions of the library.

#### 6. Test utility V7 naming leftovers

- `expectFieldValueV7` in assertions.ts — used in ~50 test files. There's no `expectFieldValueV6` anymore. Could be renamed to `expectFieldValue` (low-priority, big diff).
- `v7Response` variable names in testControlledUnControlled.tsx — remnants from when there were separate v6/v7 test paths. Could be renamed to just `response`.

---

### Nice-to-haves (not blocking)

#### 7. Demo files still named "V7"

`BrowserV7Field.tsx`, `BrowserV7SingleInputRangeField.tsx`, `BrowserV7MultiInputRangeField.tsx`, and `MaterialV7FieldWrapped.tsx` all have "V7" in their names with no V6 counterpart. Renaming to drop "V7" would improve clarity, but this changes URLs and is a bigger follow-up.

#### 8. Codemod README could mention spread-prop limitation

The codemod README doesn't note that spread props (`<DatePicker {...props} />`) won't be handled. The migration guides do mention it generally, but a brief note in the codemod section would be helpful.

---

### Things done well

- `getSectionOrder` properly simplified (removed `isRtl` for V6 path)
- `buildSectionsFromFormat` RTL handling correctly simplified — always applies the V7 reversal now
- `useParsedFormat` properly passes `isRtl: false` with a clear comment instead of using `enableAccessibleFieldDOMStructure: false` hack
- `createDateStrForHiddenInputFromSections` renamed correctly (dropped "V7")
- All generic `TEnableAccessibleFieldDOMStructure` parameters removed consistently
- The deleted `useFieldV6TextField.ts` (569 lines) was the right call
- Codemod is correct and well-tested for its scope

File: docs/data/date-pickers/custom-field/BrowserField.js
Comment (LukasTy) [inline docs/data/date-pickers/custom-field/BrowserField.js:1]: Sadly it didn't detect that this got renamed like the original `tsx` file, probably because it's rebuilt with `docs:typescript:formatted`.

**Comment by LukasTy** [comment]
> #### 5. Duplicate error code in error-codes.json
> The error about `sectionListRef` not being initialized now exists in two codes:
> 
> * error-codes.json — old message, still suggests `enableAccessibleFieldDOMStructure={false}` (a prop that no longer exists)
> * error-codes.json — new message, correctly updated
> 
> Per the project guidelines: _"If the update created a new error code, but the new and original message have the same number of arguments and semantics haven't changed, update the original error in error-codes.json instead of creating a new code."_ The semantics did change (removed one paragraph), so having a new code (205) is arguably correct for backward compatibility. But it's worth verifying this is intentional. The old code 159 will still be served to users hitting the error on older versions of the library.

The new message: https://github.com/mui/mui-x/pull/21966/changes#diff-b3a58f32230902c186b603e587bf088503d033a4481a1ee53d03cbd8ebbfbbacR206

@brijeshb42 Does this look correct? The new message should be created if a message already in production has changed?

**Comment by LukasTy** [comment]
## Follow-up Review

### Status of my original review items

| # | Issue | Status |
|---|-------|--------|
| 1 | Stale TODO in useFieldRootProps.ts | **Fixed** — handler inlined, TODO removed |
| 2 | Outdated JSDoc comments (4 files) | **Fixed** — all "non-accessible DOM structure" references removed from hook JSDoc |
| 3 | PickerFieldUI comment | **Partially fixed** — a follow-up TODO was added (line 422), but the comment on line 421 still says "Once the non-accessible DOM structure **will be** removed" (future tense). Should be past tense or removed. |
| 4 | `useFieldV7TextField` naming | **Fixed** — file deleted, code inlined into useField.ts |
| 5 | Duplicate error code | Not addressed, and that's fine — old code 159 serves old library versions |
| 6 | `expectFieldValueV7` / `v7Response` naming | **Fixed** — both renamed |
| 7 | Demo files named "V7" | **Partially fixed** — see below |

### New issues found

#### 1. Orphaned `BrowserV7*` files not deleted

The `.tsx` files were renamed (`BrowserV7Field.tsx` → `BrowserField.tsx`, etc.), but the old companion `.js` and `.tsx.preview` files were not deleted. These 6 files are still tracked in git and serve no purpose:

- BrowserV7Field.js
- BrowserV7Field.tsx.preview
- BrowserV7MultiInputRangeField.js
- BrowserV7MultiInputRangeField.tsx.preview
- BrowserV7SingleInputRangeField.js
- BrowserV7SingleInputRangeField.tsx.preview

#### 2. `MaterialV7FieldWrapped` not renamed

`BrowserV7*` demos were renamed to drop "V7", but MaterialV7FieldWrapped.tsx (and its `.js`/`.tsx.preview` companions) were missed. The docs still reference `MaterialV7FieldWrapped.js` on custom-field.md.

#### 3. PickerFieldUI comment still uses future tense

PickerFieldUI.tsx:
```ts
/**
 * ...
 * Once the non-accessible DOM structure will be removed, we will be able to remove the `textField` slot...
 * TODO: Address with the needed support for the `textField` slotProps given the change of minimum version of MUI.
 */
```
Line 421 should be updated — the structure **has** been removed. The TODO is good, but the context above it is stale.

#### 4. Duplicate `createSectionContentProps` call (pre-existing, low-priority)

In useField.ts, `createSectionContentProps(section, sectionIndex)` is called twice per section — once to capture `content` (used for `data-range-position`), and again for the `content` property of the returned element. The second call on line 200 should just reuse the `content` variable:

```diff
      const content = createSectionContentProps(section, sectionIndex);
      return {
        container: createSectionContainerProps(sectionIndex),
-       content: createSectionContentProps(section, sectionIndex),
+       content,
```

This was pre-existing in the old useFieldV7TextField.ts, but since the code was moved, worth fixing.

#### 5. Theme augmentation comments still say "V7 Picker's TextField"

Three comments in props.ts, overrides.ts, and themeAugmentation.spec.ts — can be dropped to just "Picker's TextField" (also fix the typo "Pickers's" in the spec file). 

Completed: *Provide follow-up review* (5/5)

**Comment by LukasTy** [comment]
## Final Review Summary

### All previous issues resolved

| Issue | Status |
|-------|--------|
| Stale TODO (inline key handler) | Fixed — inlined into `useFieldRootProps` |
| Outdated JSDoc comments (4 files) | Fixed — all removed |
| `useFieldV7TextField` naming | Fixed — inlined into useField.ts, file deleted |
| `expectFieldValueV7` / `v7Response` renames | Fixed |
| Orphaned `BrowserV7*` files | Fixed — deleted |
| `MaterialV7FieldWrapped` → `MaterialFieldWrapped` | Fixed — renamed |
| Broken demo reference in custom-field.md | Fixed — updated to `MaterialFieldWrapped.js` |
| `// V7 Picker's TextField` comments | Fixed — all 3 updated |
| Duplicate `createSectionContentProps` call | Fixed — reuses `content` variable |

### Remaining (non-blocking)

- PickerFieldUI.tsx: stale future-tense comment about removing the non-accessible DOM structure. The TODO on line 422 contextualizes it as a known follow-up. Not a blocker.

### Merge-readiness score: **9.5/10**

The PR is ready to merge. The core refactor is thorough and correct — v6 code paths, types, generics, conditional logic, test utilities, demos, and documentation have all been properly cleaned up. The single remaining stale comment is cosmetic and doesn't affect functionality or developer understanding.

File: packages/x-date-pickers-pro/src/internals/models/fields.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ import type {
  */
 export interface BaseMultiInputFieldProps
   extends
-    Pick<UseFieldInternalProps<PickerRangeValue, boolean, unknown>, 'readOnly' | 'autoFocus'>,
+    Pick<UseFieldInternalProps<PickerRangeValue, boolean>, 'readOnly' | 'autoFocus'>,
     MultiInputFieldRefs {
```
[the remark this answers is not in the document]
Comment (LukasTy) [reply packages/x-date-pickers-pro/src/internals/models/fields.ts:18]: Fixed.

File: AGENTS.md [hunk 2]
```diff
@@ -137,6 +137,9 @@ To generate `.js` files, run `pnpm docs:typescript:formatted`.
 
 Codemods are run by consumers of the MUI X libraries to migrate to newer versions of the libraries.
 
+When adding a new codemod, make sure to also document it in the `x-codemod/README.MD` following existing examples.
```
[the remark this answers is not in the document]
Comment (LukasTy) [reply AGENTS.md]: Fixed

File: packages/x-date-pickers-pro/src/internals/utils/createMultiInputRangeField/createMultiInputRangeField.types.ts [hunk 3]
```diff
@@ -55,7 +52,7 @@ export interface MultiInputRangeFieldSlots {
   /**
    * Form control with an input to render a date.
    * It is rendered twice: once for the start date and once for the end date.
-   * @default <PickersTextField />, or <TextField /> from '@mui/material' if `enableAccessibleFieldDOMStructure` is `false`.
+   * @default <PickersTextField />
    */
   textField?: React.ElementType;
   /**
```
[the remark this answers is not in the document]
Comment (LukasTy) [reply packages/x-date-pickers-pro/src/internals/utils/createMultiInputRangeField/createMultiInputRangeField.types.ts:57]: Address it.

**Comment by brijeshb42** [comment]
@LukasTy  The old message pertains to the old version of the library.

So creating the new code is correct here.

File: docs/public/static/error-codes.json [hunk 4]
```diff
@@ -202,5 +202,6 @@
   "201": "MUI X: Unimplemented: rowspan feature is required",
   "202": "MUI X Charts: Series with ID \"%s\" should have an x-axis of type \"band\". Candlestick charts require a band scale for the x-axis. Set the scaleType to \"band\" for this axis.",
   "203": "MUI X Charts: OHLC series with id='%s' has no data.\nEither provide a data property to the series or use the dataset prop.",
-  "204": "MUI X Charts: OHLC series with id='%s' has incomplete datasetKeys.\nProperties %s are missing."
+  "204": "MUI X Charts: OHLC series with id='%s' has incomplete datasetKeys.\nProperties %s are missing.",
+  "205": "MUI X: The `sectionListRef` prop has not been initialized by `PickersSectionList`\nYou probably tried to pass a component to the `textField` slot that contains an `<input />` element instead of a `PickersSectionList`.\n\nLearn more about the field accessible DOM structure on the MUI documentation: https://mui.com/x/react-date-pickers/fields/#fields-to-edit-a-single-element"
```
[the remark this answers is not in the document]
Comment (LukasTy) [reply docs/public/static/error-codes.json]: This is incorrect.
Given the error-codes architecture, the old message has to be kept, since it's already used in production.

**Comment by LukasTy** [comment]
> Can you update the merge conflict and I'll review it again. Just want to make sure that the `PickerUIField` is up-to-date with master from my earlier PR.

Addressed conflicts. 👌 

___

@flaviendelangle, I've addressed valid concerns in: https://github.com/mui/mui-x/pull/21966#pullrequestreview-4055139614.

**Comment by LukasTy** [comment]
## Code Review: PR #21966 — Remove `enableAccessibleFieldDOMStructure` behavior

### Overview
Removes the `enableAccessibleFieldDOMStructure` prop and all v6 legacy field rendering code, making the v7 accessible DOM structure (section-based `PickersSectionList`) the only mode. Includes a codemod, migration guide updates, and comprehensive test/type cleanup.

Overall: well-executed removal. The diff is large but consistent. Below are edge cases and risks focused on user-facing breakage.

---

### Edge Cases & Risks

#### 1. Codemod doesn't cover `slotProps.field` usage (Medium Risk)
The codemod only strips `enableAccessibleFieldDOMStructure` as a **direct JSX prop**. Users who passed it inside `slotProps.field` objects will NOT be auto-migrated:
```tsx
// NOT handled by codemod — will cause TS error but no auto-fix
<DatePicker slotProps={{ field: { enableAccessibleFieldDOMStructure: false } }} />
```
**Suggestion:** Either extend the codemod to handle object properties inside `slotProps.field`, or explicitly call this out in the migration guide as a manual step.

#### 2. Codemod can't handle spread props (Low Risk, but worth documenting)
```tsx
const fieldProps = { enableAccessibleFieldDOMStructure: true };
<DateField {...fieldProps} />  // codemod won't catch this
```
This is inherently hard for codemods. Just note it in the migration guide.

#### 3. Removed generic type parameter `TEnableAccessibleFieldDOMStructure` (Medium Risk)
All picker/field type interfaces lose their first generic parameter. Any user code that explicitly parameterized these will break:
```tsx
// Before (compiles)
const props: DateRangePickerProps<boolean> = ...;
useDateField<true, typeof props>(props);

// After (TS error)
const props: DateRangePickerProps = ...;
useDateField<typeof props>(props);
```
No codemod covers this. The migration guide should list the affected types explicitly:
- All `*PickerProps`, `*FieldProps`, `*SlotProps` generics
- `PickerManager` (5 → 4 type params)
- `useDateField`, `useDateTimeField`, etc. hook generics

#### 4. Removed public type exports (Medium Risk)
These public exports are removed without codemod coverage:
- `UseDateManagerParameters`, `UseDateTimeManagerParameters` — deleted entirely
- `PickerManagerEnableAccessibleFieldDOMStructure` — removed from internals exports
- `createDateStrForV6InputFromSections` — removed (renamed counterpart: `createDateStrForHiddenInputFromSections`)
- `getV6InputValueFromSections` removed from `FieldValueManager` interface

Users with custom field implementations that imported these will get hard build failures. Consider adding the renamed exports to the migration guide.

#### 5. Custom `textField` slot with plain `<input>` (Medium Risk — good error, but abrupt)
Users who had `enableAccessibleFieldDOMStructure={false}` with a custom plain `<input>` in the `textField` slot will now hit error code 205 at runtime. The error message is clear, but this is a **runtime crash** rather than a compile-time error. The migration guide does mention this, but it's worth highlighting it more prominently since it's the most likely "silent v8 → v9 breakage" scenario.

#### 6. `TODO: Remove v9` comment in `PickerFieldUI.tsx` (Nit)
There's a `TODO: Remove v9 - temporary workaround` comment — but this PR *is* for v9. Should this be addressed now or relabeled?

---

### What's Done Well

- **Runtime dev warning**: Passing the removed prop emits `warnOnce` in dev mode rather than crashing — correct DX for a removed prop.
- **Error 205 safety net**: Good guard against users who passed plain `<input>` to `textField` slot (only worked with v6 mode).
- **Clean removal**: No leftover v6/v7 conditional branches in core hooks. `useField`, `buildSectionsFromFormat`, `getSectionOrder` are all cleanly simplified.
- **Test coverage**: No gaps found. V6-only tests removed; remaining tests cover the now-only accessible structure.
- **Codemod + preset-safe integration**: Properly wired into the v9.0.0 preset-safe pipeline.
- **Renames consistent**: `V7` suffix dropped from all file names, test helpers, and utility functions.

---

### Suggestions

1. **Extend codemod** to strip `enableAccessibleFieldDOMStructure` from `slotProps.field` object properties, or document it as a manual migration step.
2. **Add a "Removed Types" section** to the migration guide listing all deleted/renamed type exports.
3. **Highlight the runtime error 205 scenario** more prominently in the migration guide — users with custom plain `<input>` textField slots will crash at runtime.
4. **Resolve or relabel** the `TODO: Remove v9` in `PickerFieldUI.tsx`.

File: packages/x-date-pickers/src/internals/hooks/useField/useFieldRootHandleKeyDown.ts
Comment (LukasTy) [inline packages/x-date-pickers/src/internals/hooks/useField/useFieldRootHandleKeyDown.ts:1]: Contents directly moved to `useField/useFieldRootProps.ts`
